modules/document_repository/jsx/categoryForm.js

```diff
@@ -0,0 +1,154 @@
+import PropTypes from 'prop-types';
+import Loader from 'Loader';
+/**
+ * Document Upload Form
+ *
+ * Fetches data from Loris backend and displays a form allowing
+ * to upload a doc file
+ *
+ * @author Shen Wang
+ * @version 1.0.0
+ *
+ * */
+class DocCategoryForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      data: {},
+      formData: {},
+      uploadResult: null,
+      errorMessage: null,
+      isLoaded: false,
+    };
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.setFormData = this.setFormData.bind(this);
+    this.uploadFile = this.uploadFile.bind(this);
+    this.fetchData = this.fetchData.bind(this);
+  }
+ fetchData() {
+    return fetch(this.props.DataURL, {credentials: 'same-origin'})
+      .then((resp) => resp.json())
+      .then((data) => this.setState({data: data, isLoaded: true}))
+      .catch((error) => {
+        this.setState({error: true});
+        console.error(error);
+      });
+}
+  componentDidMount() {
+    this.fetchData();
+  }
```

Comment:
`componentDidMount` should be the first function in the class.

Reply:
done